client/src/pages/ScanStore.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {useEffect, useState} from 'react'
+import StoreHeader from './../components/StoreHeader'
+import Divider from '@material-ui/core/Divider'
+
+interface Item {
+  barcode: string;
+  name: string;
+  price: number;
+  "merchant-id": string;
+}
+
+const emptyItem = (): Item => ({
+  barcode: '',
+  name: '',
+  price: 0.0,
+  "merchant-id": ''
+})
+
+interface CartItem {
+  item: Item;
+  quantity: number;
+}
+
+const emptyCartItem = (): CartItem => ({
+  item: emptyItem(),
+  quantity: 0
+})
+
+function ScanStore() {
+  // Update URL params to find storeID
+  const curUrl = window.location.search
+  const urlParams = new URLSearchParams(curUrl)
+  const storeID = urlParams.get('id')
+
+  // Declare list of items in our cart
+  const [shoppingList, setShoppingList] = useState<CartItem[]>([])
+
+  // Get user's cart items
+  const getJson = (res: any) => {
+    let res_json = res.json()
+    if (res.ok){
+      return res_json
+    }
+    else{
+      alert(`Response code: ${res.status}`)
+      return []
+    }
+  }
+
+  const catchErr = (err: any) => {
+    alert("Error: " + err)
+  }
+
+  // fetch list of users
+  const fetchCart = async() => {
```

Comment:
consider creating a re-usable common fetch function that takes in your required params

Reply:
implemented in `utils`